eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -94,6 +94,29 @@ jobs:
   - checkout: self
     submodules: recursive
 
+  # Validate all the files are formatted correctly according to the .clang-format file.
+  - bash: |
+      # Run clang-format recursively on each source and header file within the repo sdk folder.
+      echo Check clang-formatting
+      clang-format --version
+      find sdk/ -regex '.*\.\(cpp\|hpp\)' -exec clang-format -style=file -i {} \;
+
+      if [[ `git status | grep modified | awk '{print $2}'` ]]; then
```

Comment:
Do we want untracked files to cause CI build failures? Consider borrowing aspects from the C SDK repo, such as `--untracked-files=no`:
https://github.com/Azure/azure-sdk-for-c/blob/0985be23010cb0989b591d92a57d6b276e960b92/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L344-L362

Reply:
`grep modified` will filter only modified files.  Untracked won't be there

Reply:
Using `grep` on the output from git seems brittle (because the output can change and you may not be able to catch all the changes). Use the built-in `git` command to avoid having to parse and process the output strings.

Reply:
> because the output can change

??  can you elaborate?

Reply:
In the future, Git could change the output of `git status` to use other text words instead of `modified`, or add similar for untracked/other types of changes.

Reply:
> In the future, Git could change ...

`--untracked-files=no` could also change then.  =P. Let's not worry about that

Reply:
That can't change. It's a built-in git command and removing it would be considered a breaking change. The other approach is just text output as a result of running a `git` command.

Reply:
We don't really need to worry about that.
Tell me when was the last time the output changed?

Reply:
`git status -s` and `git status --porcelain` are more efficient and are better suited for our purpose here for text processing with scripts, since git status is meant for human readability and git doesn't guarantee that won't change. For example. those options to git status don't output "modified" but rather "M".

https://www.stefanjudis.com/today-i-learned/the-short-version-of-git-status-and-the-close-but-different-porcelain-mode/
> ... the --porcelain flag is used to generate output that is easily parseable by a script. Imagine you run some automation that checks the status of a git repository before running further commands. In that case, you don't want to have help text and terminal colors included in the output.

> Additionally, **you also don't want that the output changes or differs from git version to git version. --porcelain guarantees that there won't be backward-incompatible changes to the output so that your scripts won't break with a git update.**

The same is mentioned in the documentation for git:
https://git-scm.com/docs/git-status#Documentation/git-status.txt---porcelainltversiongt
> Give the output in an easy-to-parse format for scripts. This is similar to the short output, **but will remain stable across Git versions** and regardless of user configuration.

>Tell me when was the last time the output changed?

From what I can infer, it was last changed ~7 years ago, but its probably still better to use the documented and recommended approach, since it _can_ change (i.e. git is allowed to, per their docs):
https://github.com/git/git/blame/master/wt-status.c

That said, I won't push for it any further, and leave it up to you to keep using `grep` and look for `modified` in the output of `git status`, if you feel strongly about using this approach instead.